src/ol_infrastructure/components/aws/olvpc.py

```diff
@@ -80,6 +81,8 @@ def is_private_net(cls, network: IPv4Network) -> IPv4Network:
             raise ValueError(msg)
         return network
 
+    # TODO @Ardiea: Add verification of the pod subnet and ensure
+    # that it doesn't overlap with service subnet
```

Comment:
The IPv4Network type has built-in support for network overlap checking. https://docs.python.org/3/library/ipaddress.html#ipaddress.IPv4Network.overlaps

Reply:
Yah I used it just below here in this function to ensure the service block is a subnet of the vpc. Just need to expand it further to include pod block and each other.

Reply:
👍